### PR TITLE
add logging changes

### PR DESCRIPTION
**Pull request by kuba-krj**
# Description
Add possibility to have logging heavy disabled

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary [Link to Neptune](https://app.neptune.ai/o/pmtest/org/llm-random/runs/details?viewId=9a15aa36-1063-49c4-807d-53393abb118a&shortId=LLMRANDOM-2840&type=run&lbViewUnpacked=true&sortBy=%5B%22sys%2Fcreation_time%22%5D&sortFieldType=%5B%22datetime%22%5D&sortFieldAggregationMode=%5B%22auto%22%5D&sortDirection=%5B%22descending%22%5D&suggestionsEnabled=false&query=)
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/conditional/utils/layer_manager.py [hunk 1]
```diff
@@ -44,19 +44,23 @@ def _register_layers(self, model):
                 self._layers.append((block_name, layer))
 
     def prepare_for_logging(self, step):
-        if (
-            step % self.logging_interval_light == 0
-            or step % self.logging_interval_heavy == 0
+        if step % self.logging_interval_light == 0 or (
+            self.logging_interval_heavy > 0
+            and (step % self.logging_interval_heavy == 0)
```
Comment (Simontwice) [inline research/conditional/utils/layer_manager.py:49]: no need for parentheses

File: research/conditional/utils/layer_manager.py [hunk 2]
```diff
@@ -44,19 +44,23 @@ def _register_layers(self, model):
...
             for block_name, layer in self._layers:
                 if hasattr(layer, "prepare_for_logging"):
                     layer.prepare_for_logging()
 
     def log(self, step):
         verbosity_levels = []
-        if step % self.logging_interval_heavy == 0:
+        if (self.logging_interval_heavy > 0) and (
+            step % self.logging_interval_heavy == 0
+        ):
```
Comment (Simontwice) [inline research/conditional/utils/layer_manager.py:59]: no need for parentheses except the outermost ones

File: research/conditional/utils/layer_manager.py [hunk 3]
```diff
@@ -44,19 +44,23 @@ def _register_layers(self, model):
...
             verbosity_levels = [2, 1, 0]
-        elif step % self.logging_interval_light == 0:
+        elif (self.logging_interval_light > 0) and (
+            step % self.logging_interval_light == 0
+        ):
```
Comment (Simontwice) [inline research/conditional/utils/layer_manager.py:63]: no need for parentheses except the outermost ones

**Review by Simontwice — CHANGES_REQUESTED** [review]
as in comments

File: research/conditional/utils/layer_manager.py (hunk 1 above)
Comment (crewtool) [reply research/conditional/utils/layer_manager.py:49]: Don't we need `self.logging_interval_light > 0` as well?

Comment (kuba-krj) [reply research/conditional/utils/layer_manager.py:49]: changed and added light

File: research/conditional/utils/layer_manager.py (hunk 2 above)
Comment (kuba-krj) [reply research/conditional/utils/layer_manager.py:59]: changed

File: research/conditional/utils/layer_manager.py (hunk 3 above)
Comment (kuba-krj) [reply research/conditional/utils/layer_manager.py:63]: changed

**Review by crewtool — APPROVED** [review]
Nicely done!

**Review by Simontwice — APPROVED** [review]

**Pull request merged**